robotoff/app/core.py

```diff
@@ -107,5 +138,56 @@ def save_insight(
     if insight.annotation is not None:
         return ALREADY_ANNOTATED_RESULT
 
+    if verify_annotation:
+        verified: bool = False
+
+        AnnotationVote.create(
+            insight_id=insight_id,
+            username=auth.get_username() if auth else None,
+            value=annotation,
+            device_id=device_id,
+        )
+
+        with db.atomic() as tx:
+            try:
+                existing_votes = list(
+                    AnnotationVote.select(
+                        AnnotationVote.value,
+                        peewee.fn.COUNT(AnnotationVote.value).alias("num_votes"),
+                    )
+                    .where(AnnotationVote.insight_id == insight_id)
+                    .group_by(AnnotationVote.value)
+                    .order_by(peewee.SQL("num_votes").desc())
+                )
+                insight.n_votes = functools.reduce(
+                    lambda sum, row: sum + row.num_votes, existing_votes, 0
+                )
+                insight.save()
+            except Exception as e:
+                tx.rollback()
+                raise e
+
+        # If the top annotation has more than 2 votes, consider applying it to the insight.
+        if existing_votes[0].num_votes > 2:
+            annotation = existing_votes[0].value
+            verified = True
+
+        # But first check for the following cases:
+        #  1) The 1st place annotation has >2 votes, and the 2nd place annotation has >= 2 votes.
+        #  2) 1st place and 2nd place have 2 votes each.
+        #
+        # In both cases, we consider this an ambiguous result and mark it with 'I don't know'.
+        if (
+            existing_votes[0].num_votes >= 2
+            and len(existing_votes) > 1
+            and existing_votes[1].num_votes >= 2
+        ):
+            # This code credits the last person to contribute a vote with a potentially not their annotation.
+            annotation = 0
```

Comment:
we do not seem to have an enum for this, a pity !

Reply:
Is this you wanting us to change something in the PR, or is it just a sentiment about the depressing state of the world? :)

Reply:
It was a gentle push about if you feel like fixing it in this PR (if you were already on the edge of doing it).